node/src/components/consensus.rs

```diff
@@ -357,6 +363,29 @@ where
                 let validator_changes = self.get_validator_changes();
                 responder.respond(validator_changes).ignore()
             }
+            Event::DumpState(req @ DumpConsensusStateRequest { era_id, .. }) => {
+                let requested_era = era_id.unwrap_or(self.current_era);
+
+                // We emit some log message to get some performance information and give the
+                // operator a chance to find out why their node is busy.
+                info!(era_id=%requested_era.value(), was_latest=era_id.is_none(), "dumping era via console");
+
+                let era_dump_result = self
+                    .open_eras
+                    .get(&requested_era)
+                    .ok_or_else(|| {
+                        Cow::Owned(format!(
+                            "could not dump consensus, {} not found",
+                            requested_era
+                        ))
+                    })
+                    .and_then(|era| EraDump::dump_era(era, requested_era));
+
+                match era_dump_result {
+                    Ok(dump) => req.answer(Ok(&dump)).ignore(),
+                    Err(err) => req.answer(Err(err)).ignore(),
+                }
```

Comment:
If `era_dump_result` is of `Result` type already, can it be shortened to:
```suggestion
	req.answer(era_dump_result).ignore()
```
?

Reply:
Ah, I see you can't as `asnwer` expects `Result<&EraDump>` while we have here `Result<EraDump>`.